code/modules/xenoarchaeology/xenoartifact.dm

```diff
@@ -0,0 +1,489 @@
+                            /datum/xenoartifact_trait/major/laser, /datum/xenoartifact_trait/major/corginator,
+                            /datum/xenoartifact_trait/minor/sentient, /datum/xenoartifact_trait/minor/wearable,
+                            /datum/xenoartifact_trait/major/handmore, /datum/xenoartifact_trait/major/invisible,
+                            /datum/xenoartifact_trait/major/heal), TRUE) 
+            if(!xenop.price)
+                xenop.price = pick(300, 500, 800) 
+            malfunction_mod = 8
+
+        if(BANANIUM)
+            name = "Australium [name]"
+            generate_traits(list(/datum/xenoartifact_trait/major/sing))
+            if(!xenop.price)
+                xenop.price = pick(500, 800, 1000) 
+            malfunction_mod = 0.5
+
+    icon_state = null
+    for(var/datum/xenoartifact_trait/T in traits) //This is kinda weird but it stops certain runtime cases.
+        if(istype(T, /datum/xenoartifact_trait/minor/dense))
+            T.on_init(src)
+            return
+    for(var/datum/xenoartifact_trait/T in traits)
+        T.on_init(src)
+
+    //Random sprite process, I'd like to maybe revisit this, make it a function. probably don't
+    if(!(icon_state))
+        var/holdthisplease = pick(1, 2, 3)
+        icon_state = "IB[holdthisplease]"//base
+        generate_icon(icon, "IBL[holdthisplease]", material)
+    if(pick(1, 0) || icon_slots[1])//Top
+        if(!(icon_slots[1])) //Some traits can set this too, it will be set to a code that looks like 901, 908, 905 ect.
+            icon_slots[1] = pick(1, 2)
+        generate_icon(icon, "ITP[icon_slots[1]]")
+        generate_icon(icon, "ITPL[icon_slots[1]]", material)
+
+    if(pick(1,0,0) || icon_slots[3])//Left
+        if(!(icon_slots[3]))
+            icon_slots[3] = pick(1, 2)
+        generate_icon(icon, "IL[icon_slots[3]]")
+        generate_icon(icon, "ILL[icon_slots[3]]", material)
+
+    if(pick(1, 0) || icon_slots[4])//Right
+        if(!(icon_slots[4]))
+            icon_slots[4] = pick(1, 2)
+        generate_icon(icon, "IR[icon_slots[4]]")
+        generate_icon(icon, "IRL[icon_slots[4]]", material)
+
+    if(pick(1,0,0) || icon_slots[2])//Bottom
+        if(!(icon_slots[2]))
+            icon_slots[2] = pick(1, 2)
+        generate_icon(icon, "IBTM[icon_slots[2]]")
+        generate_icon(icon, "IBTML[icon_slots[2]]", material)
+
+/obj/item/xenoartifact/examine(mob/user)
+    . = ..()
+    for(var/obj/item/clothing/glasses/science/S in user.contents) //This isn't done the best. I wonder If they'll let it sit.
+        to_chat(user, "<span class='notice'>[special_desc]</span>")
+
+/obj/item/xenoartifact/interact(mob/user)
+    . = ..()
+    if(process_type == "lit") //Snuff out candle
+        process_type = ""
+        set_light(0)
+        return
+    if(user.a_intent == INTENT_GRAB)
+        if(touch_desc)
+            touch_desc.on_touch(src, user)
+        return
```

Comment:
This is a bit difficult to explain, please probe further if I don't do this well.
Using it in hand, in cases where it's a tactile artifact, it will activate, if you have any intent that isn't help. Using the grab-intent to pick up artifacts also means it wont activate, unlike a help-intent pick up which will activate dense artifacts, which can't be picked up. This is supposed to act as if you're palming the artifact or running your hand across it.
It's a special message you can check for if it's there, without impeding on any other interactions. 
There's similar situations for using items on artifacts. If you use a help intent on a window with a lighter, you'll attack the window. In the artifact's case, you'll try to light it, even if you can't, without damaging it. Any other intent will attack it.
This just helps with the process of sleuthing being simple & firm, not risking interaction with multiple systems.